api/logger.py

```diff
@@ -0,0 +1,35 @@
+import logging
+from api.utils import running_local
+
+
+class DefaultLogger(object):
+  def __init__(self, is_running_local):
+    self.logger = logging.getLogger("pcluster-manager")
+    if is_running_local:
+      handler = logging.StreamHandler()
+      handler.setFormatter(logging.Formatter("%(asctime)s - %(levelname)s - %(message)s"))
+      self.logger.addHandler(handler)
+      self.logger.setLevel(logging.DEBUG)
+    else:
+      self.logger.setLevel(logging.INFO)
+
+  def _log_output(self, msg, extra):
+    extra["message"] = msg
+    return extra
+
+  def debug(self, msg, extra={}):
+    self.logger.debug(self._log_output(msg, extra)) 
+
+  def info(self, msg, extra={}):
+    self.logger.info(self._log_output(msg, extra))  
+
+  def warning(self, msg, extra={}):
+    self.logger.warning(self._log_output(msg, extra))   
+    
+  def error(self, msg, extra={}):
+    self.logger.error(self._log_output(msg, extra))
+
+  def critical(self, msg, extra={}):
```

Comment:
issue(blocking): the default parameter `extra` is not initialized for every call, so it will hold consecutive values added if it's not cleared after every use, you can read more [here](https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python)

Reply:
Going to follow the advice and set `None` as default